### DISCUSSION TITLE
WorkItemType

### DISCUSSION THREAD

**Discussion by iand123** [nbrown02/FlowViz-Jira#13] [Ideas]
Hi Nick

Long time azure devops user and using it for the first time in JIRA

Due to JIRA setup pain (dont we all love it) i was trying to add a slicer to select item types for the data. I couldnt get it to work exactly so i guess my question is 2 fold

1. any plans on adding this in the future?
2. in trying to add it myself i noticed that the fields used in various tables for issue type are slightly different named which is why my attempt does work. for example some fields are called IssueType and other are WorkItemType. It may not be the issue but would this affect the slicer working? If that is the issue any plans to be able to standardise that field across the different data tables?

**Comment by iand123** [comment]
I think i solved my own problem, on point 2 i added the relationship between the columns and it seems to be working

Just to elaborate on the need more, we have all sorts of different issue types that are coming through like retro actions etc that get used on the board. We also have other issue types that are used for tracking other things on a different board within the project. WHilst visibility of these things is good and the ability to see the frequency and how long they take they dont relate to the work the team does so automatically includes them in the visualisations as they exist in the same project. Hope that makes some sort of sense, its more of a how we've setup/used jira in some instances that like always ends up being the tail wagging the dog!

**Comment by nbrown02** [comment]
Hello! 

Ah great to hear you're using it across Jira now :)

Yeah the WorkItemType is me just calling it that being lazy as that's the name in Azure DevOps :D 
Wouldn't be a problem to add this feature in - give me a few days!

**Reply by iand123** [reply]
Awesome, thanks Nick. I'll keep an eye out for an update. As i put in my second update i can make it work by linking the relationships of the data together but figured its better off in the hands of you with your knowledge on this :)

**Reply by nbrown02** [reply]
Just deployed a new version - renamed that column back to IssueType and now have added the Issue Type slicer - it uses a standalone issue type table that has relationships with all the other tables, so will be a better way of slicing than what you'll have currently :)

Btw - do you use labels for identifying blockers? If so drop me an email (nicolas.brown@outlook.com) if you want to help test for me (will add you in the readme as a thank you if so 👍 )

**Reply by iand123** [reply]
Hi Nick, thanks for this, in the end i spent an afternoon beavering away to add it myself (along with also making it work across project by hacking the project parameter to be a csv list and change the queries to be in instead of equals)

Will download the latest version and take a look though. Also, blocked we are working out that we have variance, some use columns and some use labels. once ive got a more definitive view on that will get back to you regarding testing out :)
